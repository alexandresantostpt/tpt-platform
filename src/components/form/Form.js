import PropTypes from 'prop-types'
import React, { forwardRef } from 'react'
import styled from 'styled-components'

import { validate } from '@utils/form'

import { Form as FinalForm } from 'react-final-form'

const StyledForm = styled.form`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
`

const Form = forwardRef(({ children, fields, initialValues, onSubmit }, ref) => (
    <FinalForm
        initialValues={initialValues}
        onSubmit={onSubmit}
        ref={ref}
        render={({ handleSubmit }) => <StyledForm onSubmit={handleSubmit}>{children}</StyledForm>}
        validate={values => validate(values, fields)}
    />
))

Form.defaultProps = {
    initialValues: {},
    ref: {}
}

Form.propTypes = {
    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]).isRequired,
    fields: PropTypes.object.isRequired,
    initialValues: PropTypes.object,
    onSubmit: PropTypes.func.isRequired
}

export default Form
