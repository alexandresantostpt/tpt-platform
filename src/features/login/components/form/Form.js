import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const StyledForm = styled.form`
    align-items: center;
    display: flex;
    flex-direction: column;
    margin-top: 3rem;
`

const Form = ({ children, onSubmit }) => <StyledForm onSubmit={onSubmit}>{children}</StyledForm>

Form.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.func, PropTypes.object, PropTypes.string]).isRequired,
    onSubmit: PropTypes.func.isRequired
}

export default Form
