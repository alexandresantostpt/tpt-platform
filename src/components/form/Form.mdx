---
menu: Form
name: Form
route: /components/form
---
import { Playground, PropsTable } from 'docz'

import Form from './Form'
import FormContainer from './container/FormContainer'
import FormField from './field/FormField'

import '@cfg/scripts'
import '@cfg/style'

# Form

<PropsTable of={ Form }/>

## Basic usage

```
<Form/>
```

<Playground>
    <Form
        fields={ {
            clientName: { component: 'input', id: 'clientName', name: 'clientName', required: false, type: 'text' },
            clientCpf: { component: 'input', id: 'clientCpf', name: 'clientCpf', required: true, type: 'text' }
        } }
        onSubmit={ values => console.log(values) }
    >
        <FormContainer>
            <FormField { ...{ component: 'input', id: 'clientName', name: 'clientName', required: false, type: 'text' } } />
            <FormField { ...{ component: 'input', id: 'clientCpf', name: 'clientCpf', required: true, type: 'text' } } />
        </FormContainer>
    </Form>
</Playground>
