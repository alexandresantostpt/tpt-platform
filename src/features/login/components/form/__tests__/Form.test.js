import React from 'react'
import renderer from 'react-test-renderer'

import Form from '../Form'

describe('Tests for Form component', () => {
    it('Should rendering without crash', () => {
        const tree = renderer
            .create(
                <Form>
                    <h1>Must be show!</h1>
                </Form>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
