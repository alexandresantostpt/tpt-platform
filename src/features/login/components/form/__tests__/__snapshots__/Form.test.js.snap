// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Form component Should rendering without crash 1`] = `
<form
  className="Form__StyledForm-sc-2b0hvf-0 hgpEDk"
>
  <h1>
    Must be show!
  </h1>
</form>
`;
